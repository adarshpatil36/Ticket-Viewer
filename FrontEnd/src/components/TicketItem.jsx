import React from "react";
import { CONSTANTS } from "../constants/constants";
import { formateDate } from "../utils/util";

export default function TicketItem({ item, onItemClick }) {
  return (
    <div
      className="TicketItem"
      data-testid="ticketItem"
      onClick={() => onItemClick(item)}
    >
      <div>
        <span>
          <label>{CONSTANTS.TICKET_DETAILS.ID}</label>
          <span>{item.id}</span>
        </span>
        <span>
          <label>{CONSTANTS.TICKET_DETAILS.SUBJECT}</label>
          <span>{item.subject}</span>
        </span>
      </div>
      <div>
        <span>
          <label>{CONSTANTS.TICKET_DETAILS.CREATED_AT}</label>
          <span>{formateDate(item.created_at)}</span>
        </span>
        <span>
          <label>{CONSTANTS.TICKET_DETAILS.STATUS}</label>
          <span>{item.status}</span>
        </span>
      </div>
    </div>
  );
}
