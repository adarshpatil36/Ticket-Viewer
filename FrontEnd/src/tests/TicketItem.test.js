import React from "react";
import { render } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import TicketItem from "../components/TicketItem";

describe("TicketItem testing", () => {
  const mockData = {
    url: "https://zendeskcodingchallenge9546.zendesk.com/api/v2/requests/4.json",
    id: 4,
    status: "open",
    priority: null,
    type: null,
    subject: "Test ticket #2",
    description: "This is test ticket #2",
    organization_id: 1500628004841,
    via: {
      channel: "api",
      source: {
        from: {},
        to: {},
        rel: null,
      },
    },
    custom_fields: [],
    requester_id: 1909839417125,
    collaborator_ids: [],
    email_cc_ids: [],
    is_public: true,
    due_at: null,
    can_be_solved_by_me: true,
    created_at: "2021-11-20T10:48:58Z",
    updated_at: "2021-11-20T10:48:58Z",
    recipient: null,
    followup_source_id: null,
    assignee_id: 1909839417125,
    ticket_form_id: 1500003279781,
    fields: [],
  };
  it("renders TicketItem component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <TicketItem item={mockData} onItemClick={() => {}}></TicketItem>,
      div
    );
  });

  it("renders TicketItem with proper label", () => {
    const { queryByLabelText } = render(
      <TicketItem item={mockData} onItemClick={() => {}}></TicketItem>
    );
    expect(queryByLabelText).toBeTruthy();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(<TicketItem item={mockData} onItemClick={() => {}}></TicketItem>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
