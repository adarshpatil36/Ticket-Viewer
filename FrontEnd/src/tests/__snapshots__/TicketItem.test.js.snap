// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketItem testing renders correctly 1`] = `
<div
  className="TicketItem"
  data-testid="ticketItem"
  onClick={[Function]}
>
  <div>
    <span>
      <label>
        ID: 
      </label>
      <span>
        4
      </span>
    </span>
    <span>
      <label>
        Subject: 
      </label>
      <span>
        Test ticket #2
      </span>
    </span>
  </div>
  <div>
    <span>
      <label>
        Created At: 
      </label>
      <span>
        Sat Nov 20 2021
      </span>
    </span>
    <span>
      <label>
        Status: 
      </label>
      <span>
        open
      </span>
    </span>
  </div>
</div>
`;
